services/apps/alcs/src/portal/public/notice-of-intent/public-notice-of-intent.service.ts

```diff
@@ -0,0 +1,124 @@
+import { ServiceNotFoundException } from '@app/common/exceptions/base.exception';
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+import { Injectable } from '@nestjs/common';
+import { VISIBILITY_FLAG } from '../../../alcs/application/application-document/application-document.entity';
+import { NoticeOfIntentDecisionV2Service } from '../../../alcs/notice-of-intent-decision/notice-of-intent-decision-v2/notice-of-intent-decision-v2.service';
+import { NoticeOfIntentDecision } from '../../../alcs/notice-of-intent-decision/notice-of-intent-decision.entity';
+import { NoticeOfIntentDocument } from '../../../alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.entity';
+import { NoticeOfIntentDocumentService } from '../../../alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service';
+import { NoticeOfIntentService } from '../../../alcs/notice-of-intent/notice-of-intent.service';
+import { NoticeOfIntentPortalDecisionDto } from './notice-of-intent-decision.dto';
+import { NoticeOfIntentParcel } from '../../notice-of-intent-submission/notice-of-intent-parcel/notice-of-intent-parcel.entity';
+import { NoticeOfIntentParcelService } from '../../notice-of-intent-submission/notice-of-intent-parcel/notice-of-intent-parcel.service';
+import { NoticeOfIntentSubmission } from '../../notice-of-intent-submission/notice-of-intent-submission.entity';
+import { NoticeOfIntentSubmissionService } from '../../notice-of-intent-submission/notice-of-intent-submission.service';
+import {
+  PublicDocumentDto,
+  PublicParcelDto,
+} from '../application/public-application.dto';
+import { PublicNoticeOfIntentSubmissionDto } from './public-notice-of-intent.dto';
+
+@Injectable()
+export class PublicNoticeOfIntentService {
+  constructor(
+    private noticeOfIntentService: NoticeOfIntentService,
+    private noticeOfIntentSubmissionService: NoticeOfIntentSubmissionService,
+    private noticeOfIntentParcelService: NoticeOfIntentParcelService,
+    private noticeOfIntentDocumentService: NoticeOfIntentDocumentService,
+    private noticeOfIntentDecisionV2Service: NoticeOfIntentDecisionV2Service,
+    @InjectMapper() private mapper: Mapper,
+  ) {}
+
+  async getPublicData(fileNumber: string) {
+    const noticeOfIntent = await this.noticeOfIntentService.getByFileNumber(
+      fileNumber,
+    );
+
+    //Easy way to check if its public
+    if (!noticeOfIntent?.dateReceivedAllItems) {
```

Comment:
I wonder if this should be moved to a separate function "validatePublicAccess(..." or something. This way no need for comment and also it will be reusable for other notice_of_intent related methods

Reply:
I won't make that change just yet since nothing else uses it, but when I add another function with the same check Ill create the helper function as you suggest.